cluster/service_registry_test.go

```diff
@@ -0,0 +1,143 @@
+package cluster
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/etcd/client"
+	"go.etcd.io/etcd/embed"
+)
+
+var TestEtcdAddr string
+
+func TestMain(m *testing.M) {
+	addr, cleanup := startTestEtcd()
```

Comment:
import flag and run `flag.Parse()` in setup if you want to read flags in test. ex if `-v` is set